.github/workflows/blank.yml

```
@@ -1,38 +1,86 @@
 jobs:
-  # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      - uses: reviewdog/action-setup@v1
+      - name: PR head repo
+        id: head_repo_name
+        run: |
+          HEAD_REPO_NAME=$(jq -r '.pull_request.head.repo.full_name' "$GITHUB_EVENT_PATH")
+          echo "PR head repo: $HEAD_REPO_NAME"
+          echo "::set-output name=repo::$HEAD_REPO_NAME"
+
+      - name: Checkout git repo
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ steps.head_repo_name.outputs.repo }}
+          fetch-depth: 0
+
+      - uses: actions/cache@v2
+        id: cache
         with:
-          reviewdog_version: latest # Optional. [latest,nightly,v.X.Y.Z]
+          path: |
+            ./vcpkg
+          key: ${{ runner.os }}-${{ hashFiles('scripts/bootstrap*') }}
 
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - name: bootstrap
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: ./bootstrap-vcpkg.sh
 
-      - name: Run a one-line script
+      - name: Formatting
         run: |
-          ./bootstrap-vcpkg.bat
-          ./vcpkg x-format-manifest ports/*/vcpkg.json
-          ./vcpkg x-add-version --overwrite-version --all
-          TMPFILE=$(mktemp)
-          git diff >"${TMPFILE}"
-          git stash -u && git stash drop
-          reviewdog -f=diff -f.diff.strip=1 -reporter=github-pr-review < "${TMPFILE}"
+          git config user.email "none@none.none"
+          git config user.name "none"
+
+          ./vcpkg format-manifest ports/*/vcpkg.json
+          git diff >.github-pr.format-manifest
+          git add -u
+          git commit -m "tmp" --allow-empty
+          ./vcpkg x-add-version --overwrite-version --all --skip-formatting-check
+          git diff >.github-pr.x-add-version
+          git reset HEAD~ --mixed
+
+      - uses: actions/github-script@v4
+        with:
+          script: |
+            const { promises: fs } = require('fs')
+            const add_version = (await fs.readFile('.github-pr.x-add-version', 'utf8')).trim()
+            const format = (await fs.readFile('.github-pr.format-manifest', 'utf8')).trim()
+
+            var output = ''
+            if (format !== "") {
+              output += "Did you remember to format all modified manifest files?\n"
+              output += "`./vcpkg format-manifest ports/*/vcpkg.json`\n"
+              output += "```diff\n" + format + "\n```\n\n"
+            }
+            if (add_version !== "") {
+              output += "Did you remember to update the version database after committing all other changes?\n"
+              output += "`./vcpkg x-add-version --all --overwrite-versions`\n"
+              output += "```diff\n" + add_version + "\n```\n\n"
```

```suggestion
              output += "You need to update the version database after committing other changes.\n"
              output += "If you added a new version, use:\n"
              output += "`./vcpkg x-add-version --all`\n"
              output += "If you updated an existing, not yet merged to microsoft/vcpkg/master, use:\n"
              output += "`./vcpkg x-add-version --all --overwrite-version`\n"
              output += "`The changes should look something like:\n"
              output += "```diff\n" + add_version + "\n```\n\n"
```